build-tools/config/webpack/webpack.conf.base.js

```diff
@@ -0,0 +1,24 @@
+const helpers = require('./webpack.helpers');
+const config = require('../config');
+
+module.exports = isDevelopment => {
+  const generator = helpers.compose(
+    [
+      require('./webpack.partial.conf.devServer'),
+      require('./webpack.partial.conf.entry'),
+      require('./webpack.partial.conf.module'),
+      require('./webpack.partial.conf.node'),
+      require('./webpack.partial.conf.optimization'),
+      require('./webpack.partial.conf.output'),
+      require('./webpack.partial.conf.plugins'),
+      require('./webpack.partial.conf.resolve'),
+    ].map(module => module(config, isDevelopment))
```

Comment:
Should we strip down the config to only pass the `'dev'` config when `isDevelopment` is `true`, and only the `'prod'` config when `isDevelopment` is `false`? (and of course keep the generic config in)

Reply:
Should we use .env files?

Maybe move the config variables there too.

Reply:
Can you execute logic in `.env` files? Currently we have some logic in the config file (for path resolving, and checking cli args).

Having env files would split up that config partly, since some of it would remain in the config js file?

Besides that, i'm not against the idea :)

Reply:
@gianko using .env files has implications beyond webpack configuration, so I think we should discuss that in a separate ticket. Feel free to create that.  Other than that I'm also not opposed to the idea.

Reply:
@ThaNarie are you referring to for example 'devServer'? I've considered it, but I think it will mostly make the config slightly more complex / less readable. I think it's fine to keep them in, though I don't have a strong preference

Reply:
Regarding config stripping, I mean:
```
{
  dev: { dev stuff }
  prod: { prod stuff }
 ... generic stuff
}
```

Currently the whole config is passed along, and in the files it still uses config.dev or config.dist.
It seems nicer to pick the generic stuff and the current build target, and merge them to the same level, so in your configuration files you don't care anymore.

Reply:
@ThaNarie I think I have an acceptable solution now, please see this commit message: https://github.com/hjeti/vue-skeleton/pull/82/commits/a059d401b5d1eb8118577e5d9e077880afd98a69

Let me know if this is ok